tests/core/pm-module/test_ens_integration.py

```diff
@@ -0,0 +1,156 @@
+import pytest
+
+from ens import ENS
+from web3 import Web3
+from web3.exceptions import (
+    BadFunctionCallOutput,
+    PMError,
+)
+
+try:
+    from ethpm import (
+        ASSETS_DIR,
+    )
+    from web3.pm import (
+        PM,
+    )
+except ImportError:
+    ethpm_installed = False
+else:
+    ethpm_installed = True
+
+
+def bytes32(val):
+    if isinstance(val, int):
+        result = Web3.toBytes(val)
+    else:
+        raise TypeError('val %r could not be converted to bytes')
+    if len(result) < 32:
+        return result.rjust(32, b'\0')
+    else:
+        return result
+
+
+@pytest.fixture
+def ens_setup(solc_deployer):
+    ENS_MANIFEST = ASSETS_DIR / 'ens' / '1.0.1.json'
+    ens_deployer = solc_deployer(ENS_MANIFEST)
+    w3 = ens_deployer.package.w3
+
+    # ** Set up ENS contracts **
+
+    # remove account that creates ENS, so test transactions don't have write access
+    accounts = w3.eth.accounts
+    ens_key = accounts.pop()
+
+    # create ENS contract
+    eth_labelhash = w3.keccak(text='eth')
+    eth_namehash = bytes32(0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae)
+    resolver_namehash = bytes32(0xfdd5d5de6dd63db72bbc2d487944ba13bf775b50a80805fe6fcaba9b0fba88f5)
+    ens_package = ens_deployer.deploy("ENSRegistry", transaction={"from": ens_key})
+    ens_contract = ens_package.deployments.get_instance("ENSRegistry")
+
+    # create public resolver
+    public_resolver_package = ens_deployer.deploy(
+        "PublicResolver",
+        ens_contract.address,
+        transaction={"from": ens_key}
+    )
+    public_resolver = public_resolver_package.deployments.get_instance("PublicResolver")
```

Comment:
These seem like it would be beneficial to split them up into more granular fixtures, but I don't see that as required (just that it might make this easier to read/understand if it splits up well into smaller pieces.